rosbag2_transport/src/rosbag2_transport/rosbag2_transport_python.cpp

```diff
@@ -181,6 +184,19 @@ rosbag2_transport_play(PyObject * Py_UNUSED(self), PyObject * args, PyObject * k
   play_options.read_ahead_queue_size = read_ahead_queue_size;
   play_options.rate = rate;
 
+  if (topics) {
+    PyObject * topic_iterator = PyObject_GetIter(topics);
+    if (topic_iterator != nullptr) {
+      PyObject * topic;
+      while ((topic = PyIter_Next(topic_iterator))) {
```

Comment:
Can this be simplified?

Reply:
I don't know enough about Python C to simplify it with confidence that I'm not introducing errors. I copied it from the record command's processing of topics [above](https://github.com/ros2/rosbag2/blob/master/rosbag2_transport/src/rosbag2_transport/rosbag2_transport_python.cpp#L153). It seems that vectors of strings have to be treated differently when converting from Python. I don't know why it has to do `PYDECREF` as opposed to the normal way to loop...